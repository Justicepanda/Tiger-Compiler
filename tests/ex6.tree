$
	<tiger-program>
		let
		<declaration-segment>
			<type-declaration-list>
			<var-declaration-list>
				<var-declaration>
					var
					<id-list>
						id
						<id-list-tail>
							,
							id
							<id-list-tail>
					:
					<type-id>
						id
					<optional-init>
					;
				<var-declaration-list>
			<funct-declaration-list>
		in
		<stat-seq>
			<stat>
				id
				<stat-id>
					<lvalue>
					:=
					<stat-id-tail>
						<expr>
							<andorterm>
								<ineqterm>
									<addterm>
										<multterm>
											<factor>
												<const>
													INTLIT
											<multterm2>
										<addterm2>
									<ineqterm2>
								<andorterm2>
				;
			<stat-seq>
				<stat>
					while
					<expr>
						<andorterm>
							<ineqterm>
								<addterm>
									<multterm>
										<factor>
											(
											<expr>
												<andorterm>
													<ineqterm>
														<addterm>
															<multterm>
																<factor>
																	id
																	<factor-tail>
																		<lvalue>
																<multterm2>
															<addterm2>
														<ineqterm2>
															<ineq>
																<
															<addterm>
																<multterm>
																	<factor>
																		<const>
																			INTLIT
																	<multterm2>
																<addterm2>
															<ineqterm2>
													<andorterm2>
											)
										<multterm2>
									<addterm2>
								<ineqterm2>
							<andorterm2>
					do
					<stat-seq>
						<stat>
							id
