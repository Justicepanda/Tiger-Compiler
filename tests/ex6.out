LET VAR ID COMMA ID COLON ID SEMI IN ID ASSIGN INTLIT SEMI WHILE LPAREN ID LESSER INTLIT RPAREN DO ID PLUS
Parsing error (line 6):                a+ <-- expected '(' or '[' or ':='
Tiger-Compiler: Compilation unsuccessful.
